File: backend/schemas/photo.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;
const LocationSchema = require('./location');
const CameraGearSchema = require('./camera-gear');
const CameraSettingsSchema = require('./camera-settings');
const MetadataSchema = require('./metadata');
const JudgementSchema = new Schema({
  rating: { type: Number, required: true },
  inShowroom: { type: Boolean, required: true },
});

const PhotoSchema = new Schema({
  imageId: { type: String, required: true, unique: true },
  photographer: { type: String, required: true },
  title: { type: String, required: true },
  description: String,
  group: { type: String, required: false },
  taken: { type: Date, required: true },
  location: { type: LocationSchema, required: true },
  gear: { type: CameraGearSchema, required: true },
  cameraSettings: { type: CameraSettingsSchema, required: true },
  metadata: { type: MetadataSchema, required: true },
  judgement: { type: JudgementSchema, required: true },
});

module.exports = mongoose.model('Photo', PhotoSchema);

File: backend/photo/entities/src/main/scala/backend/photo/entities/Photo.scala
package backend.photo.entities

import backend.photo.entities.gear.Gear
import backend.photo.entities.meta.{Judgement, Location, Metadata}
import backend.photo.entities.settings.CameraSettings

import java.time.Instant

case class Photo(imageId: String,
                 title: String,
                 description: Option[String],
                 photographer: String,
                 group: Option[String],
                 location: Location,
                 taken: Instant,
                 gear: Gear,
                 cameraSettings: CameraSettings,
                 metadata: Metadata,
                 judgement: Judgement)
